modules/apps/transfer/keeper/msg_server.go

```diff
@@ -59,3 +66,33 @@ func (k Keeper) UpdateParams(goCtx context.Context, msg *types.MsgUpdateParams)
 
 	return &types.MsgUpdateParamsResponse{}, nil
 }
+
+// unwindHops unwinds the hops present in the tokens denomination and returns the message modified to reflect
+// the unwound path to take. It ensures only a single token must be present in the tokens list
+// and that the token is not native to the chain.
+func (k Keeper) unwindHops(ctx sdk.Context, msg *types.MsgTransfer) (*types.MsgTransfer, error) {
```

Comment:
If we pass the `msg` as a pointer, could we modify in place instead of returning it again?

Reply:
I personally think it's ok to leave it as it is. It makes it easier it's a different message and makes some code more readable, but I'm open to changing it if you feel strongly about it